fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/jobparameterprovider/JobParameterProvider.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.jobs.service.jobparameterprovider;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.fineract.infrastructure.jobs.data.JobParameterDTO;
+import org.springframework.batch.core.JobParameter;
+
+public interface JobParameterProvider {
+
+    Map<String, JobParameter> provide(List<JobParameterDTO> jobParameterDTOList);
+
+    String getJobName();
+
+    default boolean canProvideParametersForJob(String jobName) {
```

Comment:
Well. :))))
Would be nicer to put this logic into an abstract class and use the template method pattern but works this way too. It's just abusing the default methods a little bit :)))